bazel/repository_locations.bzl

```diff
@@ -218,12 +218,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "c-ares",
         project_desc = "C library for asynchronous DNS requests",
         project_url = "https://c-ares.haxx.se/",
-        version = "1.19.1",
-        sha256 = "321700399b72ed0e037d0074c629e7741f6b2ec2dda92956abe3e9671d3e268e",
+        version = "1.20.0",
+        sha256 = "cde8433e9bf6c6a0d9e7e69947745ee649256d76009d6c23b9555f84c5c13988",
         strip_prefix = "c-ares-{version}",
         urls = ["https://github.com/c-ares/c-ares/releases/download/cares-{underscore_version}/c-ares-{version}.tar.gz"],
         use_category = ["dataplane_core", "controlplane"],
-        release_date = "2023-05-22",
+        release_date = "2023-10-07",
```

Comment:
The latest `c-ares` is `1.28.1` - https://github.com/c-ares/c-ares/releases/tag/cares-1_28_1

This now closed PR outlines some of the changes required. It would be good to catch up this dependency to the latest

Reply:
@moderation  / @htuch  - is it alright if we upgrade to `1.28.1` with a follow up PR. Intention to upgrade to 1.20 was to help unblock [this](https://github.com/envoyproxy/envoy/pull/33551) PR for which 1.20 is the min requirement .